src/DependencyInjection/Compiler/HttpClientRequestIdPass.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DR\SymfonyRequestId\DependencyInjection\Compiler;
+
+use DR\SymfonyRequestId\Http\RequestIdAwareHttpClient;
+use DR\SymfonyRequestId\RequestIdStorageInterface;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Parameter;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @codeCoverageIgnore - This is a config class
+ */
+class HttpClientRequestIdPass implements CompilerPassInterface
+{
+    /**
+     * @SuppressWarnings(PHPMD.UnusedLocalVariable)
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if ($container->getParameter('digital_revolution.symfony_request_id.http_client.enabled') === false) {
```

Comment:
use const and check with $container->hasParameter first. This might throw exception when key doesnt exist.